### PR TITLE
Adventure 4.9.1

### LINKED ISSUE #6622: update adventure to 4.9.1

### PR DESCRIPTION
**Pull request by kezz**
This PR updates Paper to Adventure 4.9.1.

Included changes:
- Update to Adventure 4.9.1 (new `sendTitlePart` method implementation).
- Add `TriState` permission check to `Permissable`.
- Implement Adventure's Pointers system.
- Add Checker annotations back (removed in Adventure 4.9.0).
- Deprecates the string-based title API methods.

### CODE REVIEW

File: patches/api/0002-Build-system-changes.patch [hunk 1]
```diff
@@ -24,7 +24,16 @@ index 6e64a444fb20aabaabfb15a30d645702c11c2da8..db6a7b2e60da0d96ef96545a781f1e05
      compileOnly(annotations)
      testCompileOnly(annotations)
  
-@@ -61,7 +63,7 @@ tasks.withType<Javadoc>().configureEach {
++    // Paper start - add checker
++    val checkerAnnotations = "org.checkerframework:checker-qual:3.18.0"
++    compileOnly(checkerAnnotations)
```
Comment (Proximyst) [inline patches/api/0002-Build-system-changes.patch]: `compileOnlyApi`

File: patches/server/0011-Adventure.patch [hunk 2]
```diff
@@ -2074,6 +2105,37 @@ index f3cb4102ab223f379f60dac317df7da1fab812a8..b41452c3ed833f0e6c3b03a56fe5773e
...
+     private final PermissibleBase perm;
++    private net.kyori.adventure.pointer.Pointers adventure$pointers; // Paper - implement pointers
+ 
+     public ServerCommandSender() {
+         if (this instanceof CraftBlockCommandSender) {
+@@ -134,4 +135,18 @@ public abstract class ServerCommandSender implements CommandSender {
+         return this.spigot;
+     }
+     // Spigot end
++
```
Comment (Proximyst) [inline patches/server/0011-Adventure.patch]: No trailing spaces, please. Common git configs will fuck with this.

**Review by Proximyst — CHANGES_REQUESTED** [review]

**Review by Proximyst — APPROVED** [review]

**Review by kashike — APPROVED** [review]

File: patches/api/0007-Adventure.patch [hunk 3]
```diff
@@ -3539,6 +3558,45 @@ index 4dba721aefe4fc6699b3b4bfa7ecb0b19c2a2a1a..01dec2c877df58c9dc22445e8b1f9ce2
...
+     public Set<PermissionAttachmentInfo> getEffectivePermissions();
++
++    // Paper start - add TriState permission checks
++    /**
++     * Checks if this object has a permission set and, if it is set, the value of the permission.
++     *
++     * @param permission the permission to check
++     * @return a tri-state of if the permission is set and, if it is set, it's value
++     */
++    default net.kyori.adventure.util.@NotNull TriState checkPermission(final @NotNull Permission permission) {
```
Comment (zml2008) [inline patches/api/0007-Adventure.patch]: I'd call this `permissionValue`, but it's not hugely important. what do others think?

Comment (kezz) [reply patches/api/0007-Adventure.patch]: I went for check as the pointer is called a checker, but the method on the checker is `value`. imo either one is fine tbh

Comment (kezz) [reply patches/api/0007-Adventure.patch:3579]: I went for check as the pointer is called a checker, but the method on the checker is `value`. imo either one is fine tbh

**Comment by kezz** [comment]
Also, this PR closes #6622.

**Review by jpenilla — APPROVED** [review]

**Pull request merged**